qt/applications/workbench/workbench/plotting/figureinteraction.py

```diff
@@ -144,6 +153,9 @@ def on_scroll(self, event):
         self.redraw_annotations()
         event.canvas.draw()
 
+        """Re-enable the legend's draggability after the scroll event finishes"""
+        legend_set_draggable(legend, True)
```

Comment:
I would delete these lines and put this inside the if statement I suggested above